ciao-cli/quotas.go

```diff
@@ -0,0 +1,218 @@
+
+Updates the quota entry for the supplied tenant
+
+The update flags are:
+
+`)
+	cmd.Flag.PrintDefaults()
+	os.Exit(2)
+}
+
+func (cmd *quotasUpdateCommand) parseArgs(args []string) []string {
+	cmd.Flag.StringVar(&cmd.name, "name", "", "Name of quota or limit")
+	cmd.Flag.StringVar(&cmd.value, "value", "", "Value of quota or limit")
+	cmd.Flag.StringVar(&cmd.tenantID, "for-tenant", "", "Tenant to update quota for")
+	cmd.Flag.Usage = func() { cmd.usage() }
+	cmd.Flag.Parse(args)
+	return cmd.Flag.Args()
+}
+
+func (cmd *quotasUpdateCommand) run(args []string) error {
+	if cmd.name == "" {
+		errorf("Missing required -name parameter")
+		cmd.usage()
+	}
+
+	if cmd.value == "" {
+		errorf("Missing required -value parameter")
+		cmd.usage()
+	}
+
+	if cmd.tenantID == "" {
+		errorf("Missing required -for-tenant parameter")
+		cmd.usage()
+	}
+	var v int
+	if cmd.value == "unlimited" {
+		v = -1
+	} else {
+		var err error
+		v, err = strconv.Atoi(cmd.value)
+		if err != nil {
+			fatalf(err.Error())
+		}
+	}
+
+	req := types.QuotaUpdateRequest{
+		Quotas: []types.QuotaDetails{{
+			Name:  cmd.name,
+			Value: v,
+		}},
+	}
+
+	b, err := json.Marshal(req)
+	if err != nil {
+		fatalf(err.Error())
+	}
+
+	body := bytes.NewReader(b)
+
+	url, err := getCiaoQuotasResource()
+	if err != nil {
+		fatalf(err.Error())
+	}
+
+	ver := api.TenantsV1
+
+	url = fmt.Sprintf("%s/%s/quotas", url, cmd.tenantID)
+	resp, err := sendCiaoRequest("PUT", url, nil, body, &ver)
+	if err != nil {
+		fatalf(err.Error())
+	}
+
+	if resp.StatusCode != http.StatusCreated {
+		fatalf("Update quotas failed: %s", resp.Status)
+	}
+
+	fmt.Printf("Update quotas succeeded\n")
+
+	return nil
+}
+
+type quotasListCommand struct {
+	Flag     flag.FlagSet
+	template string
+	tenantID string
+}
+
+func (cmd *quotasListCommand) usage(...string) {
+	fmt.Fprintf(os.Stderr, `usage: ciao-cli [options] quotas list [flags]
+
+Show all quotas for current tenant or supplied tenant if admin
+
+The list flags are:
+
+`)
+	cmd.Flag.PrintDefaults()
+	fmt.Fprintf(os.Stderr, `
+The template passed to the -f option operates on a 
+
+%s`,
+		generateUsageUndecorated([]types.QuotaDetails{}))
+	fmt.Fprintln(os.Stderr, templateFunctionHelp)
+
+	os.Exit(2)
+}
+
+func (cmd *quotasListCommand) parseArgs(args []string) []string {
+	cmd.Flag.Usage = func() { cmd.usage() }
+	cmd.Flag.StringVar(&cmd.tenantID, "for-tenant", "", "Tenant to get quotas for")
+	cmd.Flag.StringVar(&cmd.template, "f", "", "Template used to format output")
+	cmd.Flag.Parse(args)
+	return cmd.Flag.Args()
+}
+
+// change this command to return different output depending
+// on the privilege level of user. Check privilege, then
+// if not privileged, build non-privileged URL.
+func (cmd *quotasListCommand) run(args []string) error {
+	url, err := getCiaoQuotasResource()
+	if err != nil {
+		fatalf(err.Error())
+	}
+
+	if cmd.tenantID != "" {
+		url = fmt.Sprintf("%s/%s/quotas", url, cmd.tenantID)
```

Comment:
I guess if a user is not privileged, this call will result in a 404 error. It might be a bit confusing. I wonder if we should check privilege ahead of time in this case so that a better error message could be generated.